### DISCUSSION TITLE
Dynamically Generate Routes Object

### DISCUSSION THREAD

**Discussion by doeixd** [vikejs/vike#940] [Discussion] [answered]
Hello! I am attempting to create a top-level `+config.js` file that dynamically specifies all the routes for my application, based upon my file structure to replicate similar functionality to SvelteKit or Solid Start nested routing.

I tried using the `fs` module in `pages/+config.js` to read my file structure, but it doesn't seem like it's possible to use this module in the `+config.js` file. Unless I'm missing something.

I know there was talk about a single route file, and the V1 design is still being worked on, so I could be missing something obvious here.

Basically, how would you attempt to replicate nested routing functionality, with built-in support for layout components? Would dynamically generating the pages object work? Is there already a mechanism for this? Would a vite plugin be needed? Any help to point me in the right direction would be greatly appreciated.

Thanks.

**Comment by brillout** [comment, marked as answer]
Yes, see https://github.com/brillout/vite-plugin-ssr/issues/341 and https://github.com/brillout/vite-plugin-ssr/issues/346.

In the meantime you can use the workaround described at https://vite-plugin-ssr.com/layouts#nested-layouts. If it doesn't work let me know why it doesn't.

> dynamically generating the pages object

Such flexibility it out-of-scope. That said, I'm open to revisit, especially given some sponsorship.
